keps/sig-auth/3926-handling-undecryptable-resources/README.md

```diff
@@ -0,0 +1,952 @@
+
+### Non-Goals
+
+<!--
+What is out of scope for this KEP? Listing non-goals helps to focus discussion
+and make progress.
+-->
+- implementing system for ignoring different types of storage errors
+
+## Proposal
+
+<!--
+This is where we get down to the specifics of what the proposal actually is.
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation. What is the desired outcome and how do we measure success?.
+The "Design Details" section below is for the real
+nitty-gritty.
+-->
+
+Improve resource retrieval errors to include more information about the object that
+failed transformation while it was being retrieved from the storage.
+
+Introduce a new `DeleteOption` that would allow deleting a resource even
+if we cannot retrieve its data.
+
+### User Stories (Optional)
+
+<!--
+Detail the things that people will be able to do if this KEP is implemented.
+Include as much detail as possible so that people can understand the "how" of
+the system. The goal here is to make this feel real for users without getting
+bogged down.
+-->
+
+#### Story 1
+
+I accidentally removed my encryption key but only a few resources were encrypted
+with it. I know that these will either be recreated by a controller, or I can
+manually recreate them. I would like a simple way to figure out which resources
+fail decryption and I would like a way to remove them via Kubernetes API.
+
+#### Story 2
+
+I would like to remove a namespace I no longer need.  However, some of the resources
+inside of the namespace were encrypted before the encryption at
+rest configuration broke, which blocks a successful namespace delete.
+
+### Notes/Constraints/Caveats (Optional)
+
+<!--
+What are the caveats to the proposal?
+What are some important details that didn't come across above?
+Go in to as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+-->
+An unconditional delete may break garbage collection, would ignore finalizers and
+would disregard any underlying system processes that might be tied to the given resource
+(e.g. pods).
+
+### Risks and Mitigations
+
+<!--
+What are the risks of this proposal, and how do we mitigate? Think broadly.
+For example, consider both security and how this will impact the larger
+Kubernetes ecosystem.
+
+How will security be reviewed, and by whom?
+
+How will UX be reviewed, and by whom?
+
+Consider including folks who also work outside the SIG or subproject.
+-->
+We need to make sure that a user that tries to perform an unconditional delete is well
+informed about the impact of what they are doing. This should be handled
+by one or more prompts from the `kubectl` client. In case of a plain HTTP DELETE
+request, at least issue a warning about what the impact of the action might be.
+Gate with an additional admission layer.
```

Comment:
Can you add here that only a specific error coming from a decrypting transformer will allow the deletion? As one of the mitigations, because I assume that temporary read errors should not ever allow that to happen.

Reply:
I'm going to reword it a bit so that it's clearer when the prompts are expected. I'll also remove the `In case of a plain HTTP DELETE request, at least issue a warning about what the impact of the action might be.` as that just seems confusing.